src/hotspot/share/utilities/elfFile.hpp

```diff
@@ -483,7 +483,7 @@ class DwarfFile : public ElfFile {
 
     DwarfFile* _dwarf_file;
     MarkedDwarfFileReader _reader;
-    uint32_t _section_start_address;
+    uintptr_t _section_start_address;
```

Comment:
I see Dean indicated this is platform specific, but I don't know how. It sounds like this is a bug if it should be 64-bit.

Reply:
I think `uint32_t` was wrong here since it could be a larger 64-bit value on 64-bit platforms. `ElfShdr` is a `typedef` for `Elf64_Shdr` here which specifies `sh_offset` as `uint64_t`:
https://github.com/openjdk/jdk/blob/43311930107d1783b742adbe567e9bdbcb55775d/src/hotspot/share/utilities/elfFile.cpp#L747-L752

 I guess `sh_offset` was just never larger than 32-bits in practice, so it went unnoticed.